### ISSUE TITLE
[Process] Use release branches to unblock PR merges

### ISSUE THREAD

**Issue by avantgardnerio** [apache/datafusion-ballista#508]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

In my understanding, at present:

1. Ballista allows git-refs to DataFusion and arrow
2. This allows features to be rapidly added to DataFusion, then Ballista features to be based on them (i.e. ListingSchemaProvider)
3. Except when a new DataFusion point release happens (i.e. v10)
4. In this case, merging of git-ref PRs is put on hold while a release of Ballista is prepared
5. After the Ballista release is live, then git-ref PRs can get merged again, and the cycle continues

The problem with this is that from step 3-5, there's a "mutex" on master and PRs are blocked, essentially "feature freeze".

**Describe the solution you'd like**

Rather than blocking multiple PRs from many contributors, it might be preferable to always leave Ballista master pointed at git-refs of DataFusion, and instead when an RC is declared to make a release branch on which the only work done is switching from git-ref back to release version of DataFusion and any accompanying compile fixes etc.

**Describe alternatives you've considered**

Keep doing what we're doing now.

**Comment by milenkovicm** [comment]
closing this issue as i dont think its valid anymore
